Pass buf into _convert_err_msg and remove global variable

ablwr opened the pull request:
Resolve #89

carlwilson commented on fido/prepare.py:
```diff
@@ -351,7 +351,7 @@ def doByte(chars, i, littleendian, esc=True):
     c2 = '0123456789ABCDEF'.find(chars[i + 1].upper())
     if (c1 < 0 or c2 < 0):
-        raise Exception(_convert_err_msg('bad byte sequence', chars[i:i + 2], i, chars))
+        raise Exception(_convert_err_msg('bad byte sequence', chars[i:i + 2], i, chars, buf=None))
```
Is this tested? I have a horrible feeling that the `buf.getValue()` in `_convert_err_msg` will crash when `buf=None`. Something (very) like `AttributeError: 'NoneType' object has no attribute 'getvalue'`

ablwr replied:
You're right -- sorry I should have put a [WIP] flag on this!!

carlwilson replied:
No problem

ablwr commented:
Okay, I rebased and just added a new `buf` here. I did some testing and there wasn't a crash when hitting this function. I'm not saying this is the best solution, but it is a solution.
